src/js/modules/infragistics.util.js

```diff
@@ -125,6 +125,23 @@
 
 		// The dummy class constructor
 		function Class() {
+			prototype = this;
+			for (name in prop) {
+				if (prop.hasOwnProperty(name)) {
+
+					// Check if we're overwriting an existing function
+					var isFn = prototype[ name ] = doSuper && typeof prop[ name ] === "function" &&
+								typeof _super[ name ] === "function" && fnTest.test(prop[ name ]);
+					if (isFn) {
+						prototype[ name ] = makeFn(name, prop[ name ]);
+					} else if ( name === "settings" && typeof  prop[ name ] === "object") {
+						prototype[ name ] = $.extend({}, prop[ name ]);
```

Comment:
I don't like that `settings` becomes a magic property that will get extended while others will not. For example, there is an object property called `schema` in `DataSchema`. Have you tried initializing two `DataSchema`-s and see if the prototype doesn't get polluted?

One way for resolving such issues is already implemented in igGrid and it's basically culling the object props  that are defined for documentation purposes on init (features and columns for the igGrid case). While it's not a clean solution either, it's at least a more targeted fix that will likely not cause butterfly effect issues.